src/Scripting/Core/ScriptBuilder.cs

```diff
@@ -73,20 +73,13 @@ public int GenerateSubmissionId(out string assemblyName, out string typeName)
             {
                 // get compilation diagnostics first.
                 diagnostics.AddRange(compilation.GetParseDiagnostics());
-                if (diagnostics.HasAnyErrors())
-                {
-                    CompilationError(diagnostics, compiler.DiagnosticFormatter);
-                }
-
+                ThrowIfAnyCompilationErrors(diagnostics, compiler.DiagnosticFormatter);
                 diagnostics.Clear();
```

Comment:
@gafter When you want to run a script warnings should be ignored. If you building a script they should be reported. Let's keep Chuck's change as is, I'm going to follow up with an appropriate scripting API change.